### PR TITLE
feat: gw7841 implement delete from bookmark sidebar

### PR DESCRIPTION
**Pull request by mudana-grune**
4. implement "delete"
https://youtrack.weseek.co.jp/issue/GW-7841

### Result
delete

### CODE REVIEW

File: packages/app/src/components/Sidebar/Bookmarks.tsx [hunk 1]
```diff
@@ -150,7 +192,7 @@ const Bookmarks = () : JSX.Element => {
                 isReachingEnd={isReachingEnd}
               >
                 {paginationResult => paginationResult?.docs.map(data => (
-                  <BookmarksItem key={data._id} data={data} swr={swr} />
+                  <BookmarksItem key={data.page._id} page={data.page} swr={swr} />
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks.tsx:195]: swrをそのままpropsで渡すってありなんだろうか...?

[the remark this answers is not in the document]
Comment (yuki-takei) [reply packages/app/src/components/Sidebar/Bookmarks.tsx:195]: はい、その方がよいです

File: packages/app/src/components/Sidebar/Bookmarks.tsx [hunk 2]
```diff
@@ -141,6 +184,7 @@ const Bookmarks = () : JSX.Element => {
                 bookmarkedPage={currentUserBookmark}
                 onUnbookmarked={mutateCurrentUserBookmarks}
                 onRenamed={mutateCurrentUserBookmarks}
+                onDeleted={onBookmarkItemDeleted}
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks.tsx:187]: The method name is not appropriate and please fix the method name like I said before. 
if you have questions, let me know.

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/components/Sidebar/Bookmarks.tsx [hunk 3]
```diff
@@ -122,6 +146,25 @@ const Bookmarks = () : JSX.Element => {
   const { t } = useTranslation();
   const { data: isGuestUser } = useIsGuestUser();
   const { data: currentUserBookmarksData, mutate: mutateCurrentUserBookmarks } = useSWRxCurrentUserBookmarks();
+  const { open: openDeleteModal } = usePageDeleteModal();
+
+  const deleteBookmarkItem = (pageToDelete: IPageToDeleteWithMeta) => {
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks.tsx:151]: `deleteMenuItemClickHandler` should be okay.

File: packages/app/src/components/Sidebar/Bookmarks.tsx [hunk 4]
```diff
@@ -77,6 +83,23 @@ const BookmarkItem = (props: Props) => {
...
+    const pageToDelete: IPageToDeleteWithMeta = {
+      data: {
+        _id: bookmarkedPage._id,
+        revision: bookmarkedPage.revision as string,
+        path: bookmarkedPage.path,
+      },
+      meta: pageInfo,
+    };
+
+    onDeleted(pageToDelete);
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks.tsx:100]:
```suggestion
    onClickDeleteMenuItem(pageToDelete);
```

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

File: packages/app/src/components/Sidebar/Bookmarks.tsx [hunk 5]
```diff
@@ -122,6 +146,25 @@ const Bookmarks = () : JSX.Element => {
...
+
+      if (isCompletely) {
+        toastSuccess(t('deleted_pages_completely', { path }));
+      }
+      else {
+        toastSuccess(t('deleted_pages', { path }));
+      }
+      mutateCurrentUserBookmarks();
+    };
+    openDeleteModal([pageToDelete], { onDeleted: onDeletedHandler });
```
Comment (kaoritokashiki) [inline packages/app/src/components/Sidebar/Bookmarks.tsx:166]:
```suggestion
    openDeleteModal([pageToDelete], { onDeleted: pageDeletedHandler });
```

Some people might confuse removing bookmarks and page deletion so I consider that we should be declare page deletion expressly.

**Review by kaoritokashiki — CHANGES_REQUESTED** [review]

**Pull request merged**